src/parser/internal/modifiers.rs

```diff
@@ -185,138 +128,117 @@ pub fn property_group(input: Vec<(Span, TokenKind, Span)>) -> ParseResult<Proper
+            TokenKind::Private => Ok(PromotedPropertyModifier::Private(*span)),
+            TokenKind::Protected => Ok(PromotedPropertyModifier::Protected(*span)),
+            TokenKind::Public => Ok(PromotedPropertyModifier::Public(*span)),
             _ => Err(ParseError::CannotUseModifierOnPromotedProperty(
                 token.to_string(),
-                *start,
+                *span,
             )),
         })
         .collect::<ParseResult<Vec<PromotedPropertyModifier>>>()?;
 
     Ok(PromotedPropertyModifierGroup { modifiers })
 }
 
-pub fn constant_group(input: Vec<(Span, TokenKind, Span)>) -> ParseResult<ConstantModifierGroup> {
+pub fn constant_group(input: Vec<(Span, TokenKind)>) -> ParseResult<ConstantModifierGroup> {
     let mut has_final = false;
     let mut has_private = false;
 
     let modifiers = input
         .iter()
-        .map(|(start, token, end)| match token {
-            TokenKind::Protected => Ok(ConstantModifier::Protected {
-                start: *start,
-                end: *end,
-            }),
-            TokenKind::Public => Ok(ConstantModifier::Public {
-                start: *start,
-                end: *end,
-            }),
+        .map(|(span, token)| match token {
+            TokenKind::Protected => Ok(ConstantModifier::Protected(*span)),
+            TokenKind::Public => Ok(ConstantModifier::Public(*span)),
             TokenKind::Private => {
                 has_private = true;
                 if has_final {
-                    Err(ParseError::FinalModifierOnPrivateConstant(*start))
+                    Err(ParseError::FinalModifierOnPrivateConstant(*span))
                 } else {
-                    Ok(ConstantModifier::Private {
-                        start: *start,
-                        end: *end,
-                    })
+                    Ok(ConstantModifier::Private(*span))
                 }
             }
             TokenKind::Final => {
                 has_final = true;
                 if has_private {
-                    Err(ParseError::FinalModifierOnPrivateConstant(*start))
+                    Err(ParseError::FinalModifierOnPrivateConstant(*span))
                 } else {
-                    Ok(ConstantModifier::Final {
-                        start: *start,
-                        end: *end,
-                    })
+                    Ok(ConstantModifier::Final(*span))
                 }
             }
             _ => Err(ParseError::CannotUseModifierOnConstant(
                 token.to_string(),
-                *start,
+                *span,
             )),
         })
         .collect::<ParseResult<Vec<ConstantModifier>>>()?;
 
     Ok(ConstantModifierGroup { modifiers })
 }
 
-pub fn collect(state: &mut State) -> ParseResult<Vec<(Span, TokenKind, Span)>> {
-    let mut collected: Vec<(Span, TokenKind, Span)> = vec![];
+pub fn collect(state: &mut State) -> ParseResult<Vec<(Span, TokenKind)>> {
```

Comment:
this function used to call `state.stream.current()` six times, to retrieve the same token 😬 ( it's my fault )